TorqueDataConnect/resources/js/TorqueDataConnectEdit.js

```diff
@@ -0,0 +1,224 @@
+        saveButtons(
+            field,
+            text,
+            handleParagraphSaveCancel,
+            handleParagraphSave
+        )
+    );
+
+    newInput[0].style.height = `${newInput[0].scrollHeight}px`;
+    console.log($(`#${$.escapeSelector(field)} > .torque-save`).data("field"));
+};
+
+const substituteParagraphValue = (sibling, target, newValue, field) => {
+    sibling.replaceWith(`<p>${newValue}</p>`);
+    const editBtn = editButton("paragraph", field);
+    editBtn.click(handleParagraphEdit);
+    target.replaceWith(editBtn);
+};
+
+const handleParagraphSaveCancel = (e) => {
+    const target = $(e.target);
+    const sibling = $(e.target).parent().prev();
+    const newValue = target.data("original");
+    const field = target.next().data("field");
+    console.log($(`#${$.escapeSelector(field)} > .torque-save`).data("field"));
+    console.log(field);
+    substituteParagraphValue(sibling, target.parent(), newValue, field);
+};
+
+const handleParagraphSave = (e) => {
+    const target = $(e.target);
+    const sibling = $(e.target).parent().prev();
+    const newValue = sibling[0].value;
+    const field = target.data("field");
+    console.log($(`#${$.escapeSelector(field)} > .torque-save`).data("field"));
+    submitEdit(field, newValue);
+    substituteParagraphValue(sibling, target.parent(), newValue, field);
+};
+
+// Unordered list event listeners
+const handleListEdit = (e) => {
+    const clickedButton = $(e.currentTarget);
+    let dataField = $(e.currentTarget).prev();
+    let listElements = [];
+    while (dataField[0].nodeName == "UL") {
+        listElements.unshift(dataField);
+        dataField = dataField.prev();
+    }
+
+    const val = listElements.map((e) => e[0].querySelector('.editable').innerText);
+    const field = clickedButton.data("field");
+
+    for (let e of listElements) {
+        e.remove();
+    }
+
+    const newInput = textArea(val.join("\n")).add(
+        saveButtons(
+            field,
+            listElements.map(e => e.find('li').html()).join('\n'),
+            handleListSaveCancel,
+            handleListSave
+        )
+    )
+
+    clickedButton.replaceWith(newInput);
+    newInput[0].style.height = `${newInput[0].scrollHeight}px`;
+    console.log($(`#${$.escapeSelector(field)} > .torque-save`).data("field"));
+};
+
+const substituteListValue = (sibling, target, newValue, field) => {
+    let listElements = "";
+    for (let l of newValue.split("\n")) {
+        listElements += `<ul><li>${l}</li></ul>`;
+    }
+    sibling.replaceWith(listElements);
+    const btn = $(editButton("list", field));
+    btn.click(handleListEdit);
+    target.replaceWith(btn);
+};
+
+const handleListSaveCancel = (e) => {
+    const target = $(e.target);
+    const sibling = $(e.target).parent().prev();
+    const newValue = target.data("original");
+    const field = target.next().data("field");
+    console.log($(`#${$.escapeSelector(field)} > .torque-save`).data("field"));
```

Comment:
Is this just debugging, or do we want it in the final version?  I think this question can be applied to all the console.log statements in js.